### PR TITLE
PUBDEV-8099: XGBoost: Fix tree reweighting - did not work for larger frames (nChunks>1)

### PR DESCRIPTION
**Pull request by michalkurka**

### CODE REVIEW

File: h2o-algos/src/test/java/hex/tree/gbm/GBMTest.java [hunk 1]
```diff
@@ -661,6 +661,7 @@ private void checkUpdateAuxTreeWeights(GBMModel gbm, Frame frame) {
     Map<Integer, SharedTreeMojoModel.AuxInfo>[][] originalAuxInfos = visitTrees(
             gbm._output._treeKeysAux, Map.class, CompressedTree::toAuxInfos);
 
+    frame = Scope.track(ensureDistributed(frame, 16));
```
Comment (maurever) [inline h2o-algos/src/test/java/hex/tree/gbm/GBMTest.java:664]: Why 16 is the correct constant here? I am just curious...

File: h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/predict/UpdateAuxTreeWeightsTask.java [hunk 2]
```diff
@@ -36,16 +36,22 @@ public UpdateAuxTreeWeightsTask(DistributionFamily dist, DataInfo di, XGBoostMod
         if (_dist != DistributionFamily.gaussian && _dist != DistributionFamily.bernoulli) {
             throw new UnsupportedOperationException("Updating tree weights is currently not supported for distribution " + _dist + ".");
         }
+    }
+
+    private double[][] initNodeWeights() {
```
Comment (maurever) [inline h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/predict/UpdateAuxTreeWeightsTask.java:41]: :+1:

**Review by maurever — APPROVED** [review]
Looks good, thank you @michalkurka. Do you have any idea how to fix this for larger frames?

**Review by koniecsveta — APPROVED** [review]

File: h2o-algos/src/test/java/hex/tree/gbm/GBMTest.java (hunk 1 above)
Comment (michalkurka) [reply h2o-algos/src/test/java/hex/tree/gbm/GBMTest.java:664]: It guarantees that one node will have at least 3 chunks and it is a good number for this dataset (the chunks won't be empty).

**Comment by michalkurka** [comment]
> Looks good, thank you @michalkurka. Do you have any idea how to fix this for larger frames?

Large frames should be fine on its own, do I understand you correctly?

**Pull request merged**